### ISSUE TITLE
Minecraft Java 1.8 no sound

### ISSUE THREAD

**Issue by Pythonic456** [Botspot/pi-apps#440]
Minecraft Java 1.8 doesn't have any sound, and this is the output in the launcher that seems relevant, I can copy-paste the whole thing if necessary:
```[10:42:32] [Client thread/INFO]: LWJGL Version: 2.9.1
[10:42:33] [Client thread/INFO]: [OptiFine] 
[10:42:33] [Client thread/INFO]: [OptiFine] OptiFine_1.8_HD_U_I7
[10:42:33] [Client thread/INFO]: [OptiFine] Build: 20180404-104900
[10:42:33] [Client thread/INFO]: [OptiFine] OS: Linux (arm) version 5.10.17-v7l+
[10:42:33] [Client thread/INFO]: [OptiFine] Java: 1.8.0_251, Oracle Corporation
[10:42:33] [Client thread/INFO]: [OptiFine] VM: Java HotSpot(TM) Server VM (mixed mode), Oracle Corporation
[10:42:33] [Client thread/INFO]: [OptiFine] LWJGL: 2.9.1
[10:42:33] [Client thread/INFO]: [OptiFine] OpenGL: V3D 4.2, version 2.1 Mesa 19.3.2, Broadcom
[10:42:33] [Client thread/INFO]: [OptiFine] OpenGL Version: 2.1.0
[10:42:33] [Client thread/INFO]: [OptiFine] Maximum texture size: 4096x4096
[10:42:33] [Thread-7/INFO]: [OptiFine] Checking for new version
[10:42:33] [Client thread/INFO]: [Shaders] ShadersMod version: 2.4.12
[10:42:33] [Client thread/INFO]: [Shaders] OpenGL Version: 2.1 Mesa 19.3.2
[10:42:33] [Client thread/INFO]: [Shaders] Vendor:  Broadcom
[10:42:33] [Client thread/INFO]: [Shaders] Renderer: V3D 4.2
[10:42:33] [Client thread/INFO]: [Shaders] Capabilities:  2.0  2.1  -  -  - 
[10:42:33] [Client thread/INFO]: [Shaders] GL_MAX_DRAW_BUFFERS: 4
[10:42:33] [Client thread/INFO]: [Shaders] GL_MAX_COLOR_ATTACHMENTS_EXT: 4
[10:42:33] [Client thread/INFO]: [Shaders] GL_MAX_TEXTURE_IMAGE_UNITS: 16
[10:42:33] [Client thread/INFO]: [Shaders] Load ShadersMod configuration.
[10:42:33] [Client thread/INFO]: [Shaders] Shaders can not be loaded, Fast Render is enabled.
[10:42:33] [Client thread/INFO]: [Shaders] No shaderpack loaded.
[10:42:33] [Client thread/INFO]: Reloading ResourceManager: Default
[10:42:33] [Client thread/INFO]: [OptiFine] *** Reloading textures ***
[10:42:33] [Client thread/INFO]: [OptiFine] Resource packs: Default
[10:42:33] [Thread-7/INFO]: [OptiFine] Version found: I7
If on Windows, make sure to provide all of the necessary dll's as specified in the twitchsdk README. Also, make sure to set the PATH environment variable to point to the directory containing the dll's.
[10:42:33] [Client thread/ERROR]: Couldn't initialize twitch stream
[10:42:34] [Sound Library Loader/INFO]: Starting up SoundSystem...
[10:42:34] [Thread-8/INFO]: Initializing LWJGL OpenAL
[10:42:34] [Thread-8/INFO]: (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
AL lib: (EE) oss_open_playback: Could not open /dev/dsp: No such file or directory
[10:42:34] [Thread-8/ERROR]: Error in class 'LibraryLWJGLOpenAL'
[10:42:34] [Thread-8/ERROR]: Unable to initialize OpenAL.  Probable cause: OpenAL not supported.
[10:42:34] [Thread-8/WARN]: ERROR MESSAGE:
[10:42:34] [Thread-8/INFO]: Could not locate OpenAL library.
[10:42:34] [Thread-8/WARN]: STACK TRACE:
[10:42:34] [Thread-8/INFO]: org.lwjgl.openal.AL.create(AL.java:151)
[10:42:34] [Thread-8/INFO]: org.lwjgl.openal.AL.create(AL.java:102)
[10:42:34] [Thread-8/INFO]: org.lwjgl.openal.AL.create(AL.java:201)
[10:42:34] [Thread-8/INFO]: paulscode.sound.libraries.LibraryLWJGLOpenAL.init(LibraryLWJGLOpenAL.java:164)
[10:42:34] [Thread-8/INFO]: paulscode.sound.SoundSystem.CommandNewLibrary(SoundSystem.java:1576)
[10:42:34] [Thread-8/INFO]: paulscode.sound.SoundSystem.CommandQueue(SoundSystem.java:2572)
[10:42:34] [Thread-8/INFO]: paulscode.sound.CommandThread.run(CommandThread.java:121)
[10:42:34] [Sound Library Loader/WARN]: ERROR MESSAGE:
[10:42:34] [Sound Library Loader/INFO]: Could not locate OpenAL library.
[10:42:34] [Sound Library Loader/INFO]: Starting up SoundSystem...
[10:42:34] [Thread-10/INFO]: Switching to No Sound
[10:42:34] [Thread-10/INFO]: (Silent Mode)
[10:42:35] [Sound Library Loader/INFO]: Sound engine started
```
And I tried this:
`libopenal1 is already the newest version (1:1.19.1-1)`

Thanks in advance for help.

**Comment by gl91306** [comment]
is it hdmi sound?

**Comment by gl91306** [comment]
I know for a fact that bluetooth sound is very buggy especially with pulseaudio and java programs

**Comment by gl91306** [comment]
Do other applications have sound, are you using pulse or alsa, and are you on 32 bit or 64 bit.?

**Comment by Pythonic456** [comment]
Ok, I think I fixed this, and it was my fault anyway.

> is it hdmi sound?

No, I was trying to use bluetooth as my monitor doesn't have speakers.

> I know for a fact that bluetooth sound is very buggy especially with pulseaudio and java programs

This is correct, although recently the RPi Team have been working on BT sound a lot, it seems.
Almost everything for me works just fine now.
Also this question made me check for pulseaudio problems, which eventually fixed it.

> Do other applications have sound, are you using pulse or alsa, and are you on 32 bit or 64 bit.?

Yes, everything else works. Any newer version of Minecraft after 1.8 works with sound fine.
32 bit.

>Someone on Discord said they copied a file from the liwjgl3 to the liwjgl2 to get sound... Maybe @mobilegmYT knows what I'm referring to?

I looked at this, but I couldn't seem to find a file which would make a difference.

All I was doing wrong seemed to be weird BT connection issues which meant the BT speaker
wasn't playing the sound, and Minecraft was not even trying to. A full reinstall of MC and some adjustment to
pulseaudio made it work.

I guess I can close this issue now.
Thanks for your suggestions!
